### PR TITLE
osd/PG: update PGPool to detect map gaps and reset cached_removed_snaps

### PR DESCRIPTION
**Pull request by athanatos**
Fixes: http://tracker.ceph.com/issues/15943
Signed-off-by: Samuel Just <sjust@redhat.com>

### CODE REVIEW

File: src/osd/OSD.cc [hunk 1]
```diff
@@ -3000,14 +3000,9 @@ PGPool OSD::_get_pool(int id, OSDMapRef createmap)
     assert(0);
   }
 
-  PGPool p = PGPool(id, createmap->get_pool_name(id),
+  PGPool p = PGPool(createmap, id, createmap->get_pool_name(id),
```
Comment (liewegas) [inline src/osd/OSD.cc]: the third argument (name) is redundant now

Comment (liewegas) [inline src/osd/OSD.cc]: and the 4th

**Comment by athanatos** [comment]
Repushed, DNM pending another test run.

**Comment by athanatos** [comment]
Tested again: sjust@teuthology:/a/samuelj-2016-06-01_11:28:31-rados-wip-sam-testing-distro-basic-smithi/228382/remote

**Pull request merged**